src/target/ch579.c

```diff
@@ -0,0 +1,139 @@
+#include "general.h"
+#include "target.h"
+#include "target_internal.h"
+#include "cortexm.h"
+#include "adiv5.h"
+#include "buffer_utils.h"
+
+static bool ch579_wait_flash(target_s *const t, platform_timeout_s *const timeout)
+{
+    uint16_t x;
+	while (((x = target_mem_read16(t, 0x4000180a)) & 0xFF) != 0x40) {
+        DEBUG_INFO("ch579 wait %04x\n", x);
+		if (target_check_error(t))
+			return false;
+		if (timeout)
+			target_print_progress(timeout);
+	}
+	return true;
+}
+
+static bool ch579_flash_erase(target_flash_s *f, target_addr_t addr, size_t len)
+{
+    (void)len;
+    DEBUG_INFO("ch579 flash erase %08x\n", (int)addr);
+    target_mem_write32(f->t, 0x40001804, addr);
+    target_mem_write8(f->t, 0x40001808, 0xa6);
+
+    if (!ch579_wait_flash(f->t, NULL))
+        return false;
+    return true;
```

Comment:
The result of the call is already a boolean, `clang-tidy` has a lint about this exact pattern - you can just `return ch578_wait_flash(flash->t, NULL);` here, simplifying this.